cluster/images/xfn/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# TODO(negz): Download crun via the Makefile instead?
+FROM curlimages/curl@sha256:cfdeba7f88bb85f6c87f2ec9135115b523a1c24943976a61fbf59c4f2eafd78e AS curl
+
+ARG CRUN_VERSION=1.6
+ARG TARGETOS
+ARG TARGETARCH
+
+RUN curl -fSSL -o /tmp/crun https://github.com/containers/crun/releases/download/${CRUN_VERSION}/crun-${CRUN_VERSION}-${TARGETOS}-${TARGETARCH}-disable-systemd
+RUN chmod +x /tmp/crun
```

Comment:
This will break the build because `crun` doesn't work for (and thus doesn't have builds for) any OS other than Linux, and is only available for certain architectures (i.e. amd64 and aarch64). I'll need to figure out some alternative here - ideally not building xfn at all by default for OSes where it won't work.

Worth noting as well that the initial impetus to pick `crun` over `runc` no longer exists, so we could just use `runc` if need be. One reason to stick with `crun` though is that it has support for running WASM containers.